lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.ndarray.js

```diff
@@ -0,0 +1,588 @@
+	expectedISAVE = new Int32Array( [ 5, 1, 2 ] );
+
+	t.deepEqual( KASE, expectedKASE, 'returns expected value' );
+	t.deepEqual( ISGN, expectedISGN, 'returns expected value' );
+	t.deepEqual( EST, expectedEST, 'returns expected value' );
+	t.deepEqual( V, expectedV, 'returns expected value' );
+	t.deepEqual( ISAVE, expectedISAVE, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns expected values when calculating the one-norm of a square matrix', function test( t ) {
+	var expectedISAVE;
+	var expectedKASE;
+	var expectedISGN;
+	var expectedEST;
+	var expectedV;
+	var ISAVE;
+	var KASE;
+	var ISGN;
+	var work;
+	var EST;
+	var X;
+	var V;
+	var A;
+
+	A = new Float64Array([
+		1.0, 10.0, 0.0, 0.0,
+		1.0, 10.0, 0.0, 0.0,
+		1.0, -1.0, 1.0, 1.0,
+		1.0, -1.0, 1.0, 1.0
+	]);
+
+	KASE = new Int32Array( 1 );
+	EST = new Float64Array( 1 );
+	ISGN = new Int32Array( 4 );
+	ISAVE = new Int32Array( 3 );
+	X = new Float64Array( 4 );
+	V = new Float64Array( 4 );
+
+	work = new Float64Array( 4 );
+
+	while ( true ) {
+		dlacn2( 4, V, 1, 0, X, 1, 0, ISGN, 1, 0, EST, 0, KASE, 0, ISAVE, 1, 0 );
+
+		if ( KASE[ 0 ] === 0 ) {
+			break;
+		}
+		else if ( KASE[ 0 ] === 1 ) {
+			dgemv( 'no-transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 0, 0, work, 1, 0 );
+			dcopy( 4, work, 1, 0, X, 1, 0 );
+		} else if ( KASE[ 0 ] === 2 ) {
+			dgemv( 'transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 0, 0, work, 1, 0 );
+			dcopy( 4, work, 1, 0, X, 1, 0 );
+		}
+	}
+
+	expectedEST = new Float64Array( [ 22.0 ] );
+	expectedV = new Float64Array( [ 10.0, 10.0, -1.0, -1.0 ] );
+	expectedKASE = new Int32Array( [ 0 ] );
+	expectedISGN = new Int32Array( [ 1, 1, -1, -1 ] );
+	expectedISAVE = new Int32Array( [ 5, 1, 2 ] );
+
+	t.deepEqual( KASE, expectedKASE, 'returns expected value' );
+	t.deepEqual( ISGN, expectedISGN, 'returns expected value' );
+	t.deepEqual( EST, expectedEST, 'returns expected value' );
+	t.deepEqual( V, expectedV, 'returns expected value' );
+	t.deepEqual( ISAVE, expectedISAVE, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns expected values when calculating the one-norm of a square matrix (offsets)', function test( t ) {
+	var expectedISAVE;
+	var expectedKASE;
+	var expectedISGN;
+	var expectedEST;
+	var expectedV;
+	var ISAVE;
+	var KASE;
+	var ISGN;
+	var work;
+	var EST;
+	var X;
+	var V;
+	var A;
+
+	A = new Float64Array([
+		1.0, -2.0, 0.0, 0.0,
+		3.0, 4.0, -5.0, 0.0,
+		0.0, 6.0, 7.0, -8.0,
+		0.0, 0.0, 9.0, 10.0
+	]);
+
+	KASE = new Int32Array( 2 );
+	EST = new Float64Array( 2 );
+	ISGN = new Int32Array( 5 );
+	ISAVE = new Int32Array( 4 );
+	X = new Float64Array( 5 );
+	V = new Float64Array( 5 );
+
+	work = new Float64Array( 5 );
+
+	while ( true ) {
+		dlacn2( 4, V, 1, 1, X, 1, 1, ISGN, 1, 1, EST, 1, KASE, 1, ISAVE, 1, 1 );
+
+		if ( KASE[ 1 ] === 0 ) {
+			break;
+		}
+		else if ( KASE[ 1 ] === 1 ) {
+			dgemv( 'no-transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
```

Comment:
```suggestion
			// A*x:
			dgemv( 'no-transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
```

Reply:
Is `work` strictly necessary here? Can we write directly to `X` with `dgemv`?

Reply:
After looking at `dgemv`, agree with the choice to use a `work` array. Adding a comment still stands.